src/main/java/uk/gov/hmcts/divorce/noticeofchange/event/SystemApplyNoticeOfChange.java

```diff
@@ -120,17 +124,35 @@ private void updateChangeOfRepresentation(CaseDetails<CaseData, State> details,
 
         if (APPLICANT_1_SOLICITOR.getRole().equals(changeOrganisationRequest.getCaseRoleId().getRole())) {
             updateOrgPolicyAndSolicitorDetails(applicant1Solicitor,
-                    nocRequestingUser, loggedInUserEmail);
+                    nocRequestingUser, nocSolicitorOrgName, loggedInUserEmail);
+            setApplicantRepresented(details.getData().getApplicant1());
         } else {
             updateOrgPolicyAndSolicitorDetails(applicant2Solicitor,
-                    nocRequestingUser, loggedInUserEmail);
+                    nocRequestingUser, nocSolicitorOrgName, loggedInUserEmail);
+            setApplicantRepresented(details.getData().getApplicant2());
         }
     }
 
-    private static void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor,
-                                                           ProfessionalUser nocRequestingUser, String loggedInUserEmail) {
-
+    private static void updateOrgPolicyAndSolicitorDetails(
+        Solicitor applicantSolicitor, ProfessionalUser nocRequestingUser,
+        String nocRequestingUserOrgName, String loggedInUserEmail
+    ) {
         applicantSolicitor.setName(String.join(" ", nocRequestingUser.getFirstName(), nocRequestingUser.getLastName()));
         applicantSolicitor.setEmail(loggedInUserEmail);
+        applicantSolicitor.setFirmName(nocRequestingUserOrgName);
+    }
+
+    private static void setApplicantRepresented(Applicant applicant) {
+        applicant.setSolicitorRepresented(YES);
+        applicant.setOffline(NO);
```

Comment:
is this offline(NO) always the case for sols ?

Reply:
They have to be online to trigger common component NoC

Reply:
doh